js/components/accessibleButton.js

```diff
@@ -70,23 +88,38 @@ var AccessibleButton = React.createClass({
 });
 
 AccessibleButton.propTypes = {
+  autoFocus: React.PropTypes.bool,
   style: React.PropTypes.object,
   className: React.PropTypes.string,
   focusId: React.PropTypes.string,
   ariaLabel: React.PropTypes.string.isRequired,
   ariaChecked: React.PropTypes.bool,
+  ariaSelected: React.PropTypes.bool,
   ariaHasPopup: React.PropTypes.bool,
   ariaExpanded: React.PropTypes.bool,
   role: React.PropTypes.string,
+  onMouseOver: React.PropTypes.func,
+  onMouseOut: React.PropTypes.func,
+  onFocus: React.PropTypes.func,
   onClick: React.PropTypes.func
 };
 
-AccessibleButton.defaultProps = {
-  focusId: Math.random().toString(36).substr(2, 10),
-  ariaChecked: null,
-  ariaHasPopup: null,
-  ariaExpanded: null,
-  role: null,
-};
+// Define focusId as a getter so that it returns a different value
+// for each instance of AccessibleButton (defaultProps is static)
+AccessibleButton.defaultProps = Object.create({}, {
+  focusId: {
```

Comment:
I tried a few approaches for this, but couldn't make it work. We do need to have the default value before component initialization otherwise it will keep getting overwritten on render. If you have ideas on how to do this I can give it a try.